File: Rational/Rational.cpp
#include<iostream>
#include<algorithm>
using namespace std;
class Rational
{
	public:
		// Default Constructor
		Rational()
		{
			num = 1;
			deno = 1;
		}
		// Parametrized Constructor
		Rational(int a, int b)
		{
			if(b == 0)
				{
					cout<<"\nWarning : Denominator is assigned zero"<<endl;
					exit(0);
				}
			else
				Reduction(a,b);
		}
		void setNum(int x)
		{
			this->num = x;
		}
		void setDeno(int x)
		{
			this->deno = x;
		}
		int getNum()
		{
			return num;
		}
		int getDeno()
		{
			return deno;
		}
		Rational add(Rational second)
		{
			int sumNum, sumDeno;
			sumNum = this->num*second.getDeno() + this->deno*second.getNum();
			sumDeno = this->deno*second.getDeno();
			Rational sum(sumNum, sumDeno);
			return sum;
		}
		Rational sub(Rational second)
		{
			int subNum, subDeno;
			subNum = this->num*second.getDeno() - this->deno*second.getNum();
			subDeno = this->deno*second.getDeno();
			Rational sub(subNum, subDeno);
			return sub;
		}
		Rational mul(Rational second)
		{
			int mulNum, mulDeno;
			mulNum = this->num*second.getNum();
			mulDeno = this->deno*second.getDeno();
			Rational mul(mulNum, mulDeno);
			return mul;
		}
		Rational div(Rational second)
		{
			int divNum, divDeno;
			if(second.getNum() == 0)
			{
				cout<<"\nError : Division By Zero\n";
				exit(0);
			}
			divNum = this->num*second.getDeno();
			divDeno = this->deno*second.getNum();
			Rational div(divNum, divDeno);
			return div;
		}
		void printRational()
		{
			cout<<"The Rational form  is : "<< num <<"/"<< deno<<endl;
		}
		void printFractional()
		{
			float ans = (float)num / (float)deno;
			cout<<"The Fractional form is : "<< ans<<endl;
		}
	private:
		int num, deno;
		void Reduction(int a, int b)
		{
			int gcd = __gcd(a, b);
			this->num = a / gcd;
			this->deno = b / gcd;
		}
};
		}
};

File: python/Rational/rational.py

from math import gcd
class Rational:
    # default and parameterized constructor
    def __init__(self, n = 1, d = 1):
        self.num = n
        self.deno = d
        if d == 0:
            print("Denominator is assigned zero")
        else:
            self.Reduction()

    def Reduction(self):
        gcdOfTwo = gcd(self.deno, self.num)
        self.deno = int(self.deno /  gcdOfTwo)
        self.num = int(self.num /  gcdOfTwo)

    def sum(self, other):
        sumNum = self.num*other.getDeno() + self.deno*other.getNum()
        sumDeno = self.deno*other.getDeno()
        sum = Rational(sumNum, sumDeno)
        return sum

    def sub(self, other):
        subNum = self.num*other.getDeno() - self.deno*other.getNum()
        subDeno = self.deno*other.getDeno()
        sub = Rational(subNum, subDeno)
        return sub

    def mul(self, other):
        mulNum = self.num*other.getNum()
        mulDeno = self.deno*other.getDeno()
        mul = Rational(mulNum, mulDeno)
        return mul
    def div(self, other):
        divNum = self.num*other.getDeno()
        divDeno = self.deno*other.getNum()
        div = Rational(divNum, divDeno)
        return div
    def getNum(self):
        return self.num
    def getDeno(self):
        return self.deno
    def printRational(self):
        print("The Rational Form is : {}/{}".format(self.num, self.deno))
    def printFractional(self):
        fract = float(self.num) / float(self.deno)
        print("The Fractional Form is : {}".format(fract))


num1 = input("Enter the numerator for first : ")
deno1 = input("Enter the denominator for first : ")
num2 = input("Enter the numerator for denominator : ")
deno2 = input("Enter the denominator for denominator : ")

r1 = Rational(int(num1), int(deno1))
r2 = Rational(int(num2), int(deno2))
sum = r1.sum(r2)
sub = r1.sub(r2)
mul = r1.mul(r2)
div = r1.div(r2)


print("After summation :- ")
sum.printRational()
sum.printFractional()

print("After subtraction :- ")
sub.printRational()
sub.printFractional()

print("After multiplication :- ")
mul.printRational()
mul.printFractional()

print("After division :- ")
div.printRational()
div.printFractional()
